tests/ui/suggestions/as-ref-2.stderr

```diff
@@ -4,14 +4,16 @@ error[E0382]: use of moved value: `foo`
 LL |     let foo = Some(Struct);
    |         --- move occurs because `foo` has type `Option<Struct>`, which does not implement the `Copy` trait
 LL |     let _x: Option<Struct> = foo.map(|s| bar(&s));
-   |                              --- ---------------- `foo` moved due to this method call
-   |                              |
-   |                              help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
+   |                                  ---------------- `foo` moved due to this method call
 LL |     let _y = foo;
    |              ^^^ value used here after move
    |
 note: `Option::<T>::map` takes ownership of the receiver `self`, which moves `foo`
   --> $SRC_DIR/core/src/option.rs:LL:COL
+help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
+   |
+LL |     let _x: Option<Struct> = foo.as_ref().map(|s| bar(&s));
+   |                                 +++++++++
```

Comment:
I mean the intention why it wasn't the proper way in the first place, since every other case works as you suggested.